### PR TITLE
KAFKA-7031 : Connect API shouldn't depend on jersey

### PR DESCRIPTION
**Pull request by mageshn**
Connect API currently depends on Jersey API as a side-effect of KIP-285. It should only depend on the JAX RS API.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by mageshn** [comment]
@rhauch This should fix the issue KAFKA-7031. Please let me know your thoughts.

File: build.gradle [hunk 1]
```diff
@@ -1441,6 +1441,7 @@ project(':connect:basic-auth-extension') {
   dependencies {
     compile project(':connect:api')
     compile libs.slf4jApi
+    compile libs.jerseyContainerServlet
```
Comment (rhauch) [inline build.gradle]: Is this necessary as a compile dependency for the API module? If so, then it seems like something might be in API when it should be in runtime.

**Review by rhauch — COMMENTED** [review]
One question.

File: build.gradle (hunk 1 above)
Comment (mageshn) [reply build.gradle]: This is a dependency for connect:basic-auth-extension. It requires jersey implementation.

Comment (mageshn) [reply build.gradle]: Have made it testRuntime dependency

**Review by rhauch — APPROVED** [review]
LGTM. Thanks!

**Comment by mageshn** [comment]
@ewencp can you please take a look at this.

**Review by ewencp — APPROVED** [review]
Good catch, hopefully wouldn't affect many connectors (especially post-plugin.path) but better to keep unneeded dependencies out of the way.

Merging to 2.0 and trunk

**Comment by ewencp** [comment]
@mageshn also, just a nit and in this case it worked out fine, but generally we prefer filing against trunk and, if things don't cherry pick cleanly, only then do we file follow ups on older branches. not critical, just a nice thing to help standardize committer workflow.

**Comment by mageshn** [comment]
Thaks @ewencp  and sure will follow the guideline going forward.

**Pull request closed without merging**